### ISSUE TITLE
Landmark Identification

### ISSUE THREAD

**Issue by vhanded** [yangyanli/PointCNN#193]
I would like to use PointCNN to identify landmark, for the simplicity of the task, I had generated 2000 football shape (prolate spheroid) point cloud or random size with points counts from 2601 to 10201. 2 tips of the point clouds are the landmark that I am interested.

![Screenshot from 2019-10-18 15-38-01](https://user-images.githubusercontent.com/1448297/67081928-ba222b00-f1ca-11e9-94e4-d9e6ddab7a9f.png)

For illustration purpose, the 2 red points are the landmark that I am interested.

The labels are 0 and 1, with the 2 red points as 1, and other as 0.

My sampling is 2048 points, batch size of 2. Loss is improving, but almost stagnant after 500,000 iteration, or 150 epochs. mAcc is still often showing 0.5. Is that mean for every batch of training, the model only guess it right for 1 of the class, and failed for another class?

Since in every point clouds, there will be only 2 landmarks with label 1, so it means it cant identify the landmark? Or because the sampling size is too small to obtain any points which is landmark?

Thanks.

**Comment by vhanded** [comment]
After training of around 200 epochs, I test the models accuracy. 

The model sometimes get 2 dots on both ends, sometimes only 1, like this.
![Screenshot from 2019-10-20 17-43-51](https://user-images.githubusercontent.com/1448297/67157645-6dbd2380-f361-11e9-8365-a87b40f1b0df.png)

Sometimes it found 3 dots, with 2 very closely on 1 side, and 1 on another side.

Is it because of not enough training? Will it improve with more training?

Thanks.

**Comment by vhanded** [comment]
有人帮帮忙吗？Any help?

**Comment by yangyanli** [comment]
Hi @vhanded,

Your positive/negative samples are quite unbalanced, in which case, the negative samples will dominant thus the network is trend to predict almost all points into negative.

One simple solution is to increase the weight for the positive samples (the two red points). This is a widely used trick. There is no principle way for addressing unbalanced samples, as far as I know...

**Comment by vhanded** [comment]
@yangyanli  I tried your idea, it is working, just it identified several points on each end, instead of only 1 point on each end. 

I modified the cls code, to output the X,Y,Z coordinates, like this [X1, Y1, Z1, X2, Y2, Z2], it is able to predict closely to the label, but still it is not exactly on the label. 

How to output an index of the point, instead of X,Y,Z? What kind of loss function is suitable?
